RIOT-OS/RIOT
# cpu/esp32: replace RTC implementation by RTT implementation

### Contribution description

This PR replaces the RTC implementation with an RTT implementation that uses the RTC hardware timer of the ESP32. The RTC implementation used instead is the RTT based RTC implementation provided by the `rtt_rtc` module.

Although the RTC hardware module of the ESP32 is called an RTC, it is a Real Time Timer (RTT) rather than a Real Timer Clock. Therefore, the former RTC peripheral driver implementation is replaced by an RTT periphal driver implementation.

**Note:** An RTT/RTC alarm is not preserved during reboot or deep-sleep.

### Testing procedure
`tests/periph_rtc` and `tests/periph_rtc` should work
```
make BOARD=esp32-wroom-32 -C tests/periph_rtt flash test
```
```
make BOARD=esp32-wroom-32 -C tests/periph_rtc flash test
```
Furthermore, use `tests/periph_pm` can be used to test that the RTT and RTC are retained in sleep modes and during reboot.
```
CFLAGS='-DESP_PM_WUP_PINS=GPIO0 -DESP_PM_WUP_LEVEL=ESP_EXT1_WAKEUP_ALL_LOW' \
USEMODULE='shell_commands rtt_cmd' \
make BOARD=esp32-wroom-32 -C tests/periph_pm flash term
```
```
> rtc settime 2020-03-29 16:17:40
> rtc gettime
2020-03-29 16:17:43
> reboot
...
> rtc gettime
2020-03-29 16:17:49
> set 0
CPU will enter power mode 0.
```
Press `BTN0`
```
...
> rtc gettime
2020-03-29 16:18:01
```


### Issues/PRs references
